server/routes/imageUploaderAPI.js

```diff
@@ -42,13 +43,12 @@ router.post("/api/image-upload", auth, (req, res) => {
           fileName: req.file.key,
         });
         const AwsUrl = `https://${BUCKET}.s3.amazonaws.com/${req.file.key}`;
-        db.User.updateOne({ _id: req.userId }, { $set: { photoUrl: AwsUrl } })
-          .then((response) => {
-            res.send(response);
-          })
-          .catch((error) => {
-            res.status(500).send(error);
-          });
+        db.User.updateOne(
+          { _id: req.userId },
+          { $set: { photoUrl: AwsUrl } }
+        ).catch((error) => {
+          res.send(error);
```

Comment:
I think this error handling would still crash the server if it came up, maybe waiting to send the response until after updating the DB instead of before would be a better way to handle it

Reply:
Thanks for comments Dan! I did correct the error handling and also simplifed the logic in the request.